tests/compute/interface-shader-param-legalization.slang

```diff
@@ -0,0 +1,54 @@
+// interface-shader-param-legalization.slang
+
+// Test case where concrete type implementing
+// an interface has resource-type fields nested in it.
+
+//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute
+
+interface IModifier
+{
+	int modify(int val);
+}
+
+IModifier gModifier;
+
+int test(
+	int val)
+{
+	return gModifier.modify(val);
+}
+
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):dxbinding(0),glbinding(0),out
+RWStructuredBuffer<int> gOutputBuffer;
+
+[numthreads(4, 1, 1)]
+void computeMain(
+    uint3       dispatchThreadID : SV_DispatchThreadID)
+{
+    let tid = dispatchThreadID.x;
+
+    let inputVal : int = tid;
+    let outputVal = test(inputVal);
+
+    gOutputBuffer[tid] = outputVal;
+}
+
+// Now that we've define all the logic of the entry point,
+// we will define some concrete types that we can plug
+// in for the interface-type parameters.
+
+struct ConcreteData
+{
+	int offset;
+}
+
+struct ConcreteModifier : IModifier
+{
+    ConstantBuffer<ConcreteData> data;
+
+	int modify(int val) { return val + data.offset; }
```

Comment:
Tab

Reply:
Ugh. Yeah, tabs sneak into the test files quite often, because I don't have the editor I use for them set up to always insert spaces.